messaging/src/test/java/org/axonframework/eventhandling/pooled/PooledStreamingEventProcessorTest.java

```diff
@@ -851,6 +851,36 @@ void testBuildWithZeroOrNegativeBatchSizeThrowsAxonConfigurationException() {
         assertThrows(AxonConfigurationException.class, () -> builderTestSubject.batchSize(0));
         assertThrows(AxonConfigurationException.class, () -> builderTestSubject.batchSize(-1));
     }
+    @Test
+    void testReplaying() {
+        when(stubEventHandler.supportsReset()).thenReturn(true);
+        List<EventMessage<Integer>> events = IntStream.range(0, 100)
+                                                      .mapToObj(GenericEventMessage::new)
+                                                      .collect(Collectors.toList());
+        events.forEach(stubMessageSource::publishMessage);
+        mockEventHandlerInvoker();
+
+        testSubject.start();
+
+        assertWithin(1, TimeUnit.SECONDS, () -> {
+                         assertEquals(8, testSubject.processingStatus().size());
+                         assertTrue(testSubject.processingStatus().get(0).isCaughtUp());
+                         assertFalse(testSubject.processingStatus().get(0).isReplaying());
+                         assertFalse(testSubject.isReplaying());
+                     }
+        );
+        testSubject.shutDown();
+        testSubject.resetTokens(StreamableMessageSource::createTailToken);
+        testSubject.start();
+        assertWithin(1, TimeUnit.SECONDS, () -> {
+                         assertEquals(8, testSubject.processingStatus().size());
+                         assertTrue(testSubject.processingStatus().get(0).isCaughtUp());
+                         assertTrue(testSubject.processingStatus().get(0).isReplaying());
+                         assertFalse(testSubject.isReplaying());
```

Comment:
I only note the check that replaying returns false. Any possibility to test it being true for some moment in time?

What might help to make the PSEP less efficient (and thus for it to be replaying longer) is to decrease the number of segments.
You can do this by adjusting the used `testSubject` with the `createTestSubject()` and `setTestSubject()` method in this test class.

The `createTestSubject()` allows you to provide a builder for the PSEP,. where you can set the number of segments to 1 for example.

Next to this, increasing the number of published events would likely also have an impact on the replay duration. 
Thus changing that would likely increase the "window of replay" too.